pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -487,24 +519,78 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 				continue
 			}
 
-			if currentListener != nil {
+			// We checked TCP over HTTP, and HTTP over TCP conflicts above.
+			// The code below checks for TCP over TCP conflicts and merges listeners
+			if currentListenerEntry != nil {
```

Comment:
I'm not sure if this is correct.  The comment reads that this is only handling `TCP over TCP` conflicts.  However, the change above (line 408) changes the HTTP logic to set `currentListenerEntry`. IIUC, this will make this code run for HTTP listeners as well, unless I'm missing something.

Reply:
HTTP code does not enter this area at all. I set it up just for consistency.. Will change variable names

Reply:
But the code you have here would allow an HTTP listener into this code block.  I'm not sure why you think it wouldn't.  You changed the logic on line 408 ... HTTP didn't used to set the current listener.